fn_ipvoid/fn_ipvoid/components/function_ipvoid.py

```diff
@@ -0,0 +1,57 @@
+# (c) Copyright IBM Corp. 2010, 2019. All Rights Reserved.
+# -*- coding: utf-8 -*-
+# pragma pylint: disable=unused-argument, no-self-use
+"""Function implementation"""
+
+import logging
+from resilient_circuits import ResilientComponent, function, handler, StatusMessage, FunctionResult, FunctionError
+import fn_ipvoid.util.selftest as selftest
+import requests
+from fn_ipvoid.lib.resilient_common import get_url
+
+
+class FunctionComponent(ResilientComponent):
+    """Component that implements Resilient function 'fn_ipvoid"""
+
+    def __init__(self, opts):
+        """constructor provides access to the configuration options"""
+        super(FunctionComponent, self).__init__(opts)
+        self.options = opts.get("fn_ipvoid", {})
+        selftest.selftest_function(opts)
+
+    @handler("reload")
+    def _reload(self, event, opts):
+        """Configuration options have changed, save new values"""
+        self.options = opts.get("fn_ipvoid", {})
+
+    @function("fn_ipvoid")
+    def _fn_ipvoid_function(self, event, *args, **kwargs):
+        """Function: IP blacklist check, whois lookup, dns lookup, ping etc."""
+        try:
+            # Get the function parameters:
+            ip_void_request_type = self.get_select_param(kwargs.get("ip_void_request_type"))  # select, values: "IP Reputation", "Domain Blacklist", "DNS Lookup", "Email Verify", "Threat Log", "SSL Info"
+            ip_void_artifact_type = kwargs.get("ip_void_artifact_type")  # text
+            ip_void_artifact_value = kwargs.get("ip_void_artifact_value")  # text
+
```

Comment:
yield StatusMessage("Getting Intelligence from IP Void for %s:%s", ip_void_artifact_type, ip_void_artifact_value)